pkg/objstore/gcs/gcs.go

```diff
@@ -167,7 +167,11 @@ func (b *Bucket) Delete(ctx context.Context, name string) error {
 
 // IsObjNotFoundErr returns true if error means that object is not found. Relevant to Get operations.
 func (b *Bucket) IsObjNotFoundErr(err error) bool {
-	return err == storage.ErrObjectNotExist
+	eof :=  errors.Is(err, io.ErrUnexpectedEOF)
```

Comment:
I don't think it's safe to assume that a `ErrUnexpectedEOF` always mean "object doesn't exit". I think you could get an unexpected EOF even if the connection is truncated while downloading the content or maybe even other scenarios.

Reply:
Yes, I agree - and I think the google developers realize the same thing. If anything EOF should is a retriable error. I am open to other suggestions on how to get the thanks objstore code to run reliably on GCS. At the moment this work-around is getting our systems up and running at scale.

What is the level of testing that is available for different storage solutions? I saw there are some e2e tests. Does the Thanos project have environments for doing comprehensive objstore tests and collecting measurements for issues like this?